lib/src/store/mod.rs

```diff
@@ -0,0 +1,87 @@
+use std::env;
+use std::ops::Deref;
+
+use anyhow::Result;
+use clap::ValueEnum;
+
+use ostree_ext::container::OstreeImageReference;
+use ostree_ext::keyfileext::KeyFileExt;
+use ostree_ext::ostree;
+use ostree_ext::sysroot::SysrootLock;
+
+use crate::spec::ImageStatus;
+
+mod ostree_container;
+
+pub(crate) struct Storage {
+    pub sysroot: SysrootLock,
+    pub store: Box<dyn ContainerImageStoreImpl>,
+}
+
+#[derive(Default)]
+pub(crate) struct CachedImageStatus {
+    pub image: Option<ImageStatus>,
+    pub cached_update: Option<ImageStatus>,
+}
+
+pub(crate) trait ContainerImageStore {
+    fn store(&self) -> Result<Option<Box<dyn ContainerImageStoreImpl>>>;
+}
+
+pub(crate) trait ContainerImageStoreImpl {
+    fn spec(&self) -> crate::spec::Store;
+
+    fn imagestatus(
+        &self,
+        sysroot: &SysrootLock,
+        deployment: &ostree::Deployment,
+        image: OstreeImageReference,
+    ) -> Result<CachedImageStatus>;
+}
+
+impl Deref for Storage {
+    type Target = SysrootLock;
+
+    fn deref(&self) -> &Self::Target {
+        &self.sysroot
+    }
+}
+
+impl Storage {
+    pub fn new(sysroot: SysrootLock) -> Self {
+        let store = match env::var("BOOTC_STORAGE") {
```

Comment:
oh I also added this bit to set the storage via env var (for now, at least), however this appears to not work as-is.  Seems the `from_str` below always fails so it's always defaulting.

Reply:
Hmm, sanity check the `from_str` in a unit test?

Reply:
Hm seems to be because it [defaults to kebab-case](https://docs.rs/clap/latest/clap/_derive/index.html#valueenum-attributes), but I tried adding `#[value(rename_all = "camelCase")]` and it looks like it's still using kebab-case.  I think maybe it's conflicting with `#[serde(rename_all = "camelCase")]`?  Really pushing the limits of my knowledge with attribute macro magic.

Reply:
Ah yes confusing...we have both clap and serde in play here, and they operate in distinct worlds. Calling https://docs.rs/clap/latest/clap/trait.ValueEnum.html#method.from_str uses clap...so it's going to ignore whatever is configured for serde.

This has come up before, as some of the things we expose in the CLI are also serializable. 

This is a bit messy, as I personally tend to strongly prefer kebab-case for CLI things...but Kube-style APIs emphasize camelCase.

We could ensure no one wins by using `SCREAMING_SNAKE_CASE` (just kidding :smile: ))

Maybe we can add a clap alias for the camelCase names?

Reply:
Another alternative is to avoid multi-word backend names, say e.g.

```
enum Backend {
 Ostree,
 Unified
}
``` 
that will come out to the same thing in both camelCase and kebab-case? as `ostree` and `unified`?

Reply:
> Maybe we can add a clap alias for the camelCase names?

I think I like this one the best.

I must have been picking up the wrong image yesterday when I was testing this because [stripping it down in playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ffb4ff47e5f83d0cf8a40ad98664d11b), everything works fine.  That will match any case-insensive kebab or non-kebab option.